compiler/passes/convert-uast.cpp

```diff
@@ -3668,115 +3741,391 @@ Symbol* Converter::convertParam(const types::QualifiedType qt) {
   return nullptr;
 }
 
+void Converter::noteConvertedSym(const uast::AstNode* ast, Symbol* sym) {
+  if (!canScopeResolve) return;
 
-} // end anonymous namespace
+  if (symStack.size() > 0) {
```

Comment:
That's an interesting idea. Because of the way the code uses a global vs an owned right now, changing it is not so easy. However I will add a comment noting that we should do that change when we migrate gConvertedSyms to a field in the converter.